### PR TITLE
Implement bg-restart

### PR DESCRIPTION
**Pull request by CAFxX**
Also refactor and cleanup the code

The `cf push --droplet` workaround should be removed once https://github.com/cloudfoundry/cli/issues/1445#issuecomment-418987943 is solved.

### CODE REVIEW

**Comment by CAFxX** [comment]
@ArthurHlt any chance of having a look at this?

File: repo.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+	if err != nil {
+		return "", err
+	}
+	if len(d) == 0 {
+		return "", fmt.Errorf("app '%s' not found.", name)
+	}
+	return d[0], err
+}
+
+func (repo *ApplicationRepo) DoesAppExist(appName string) (bool, error) {
```
Comment (CAFxX) [inline repo.go:172]: note that this was rewritten/simplified in the process

**Comment by giner** [comment]
@CAFxX, I found a way to avoid downloading droplet https://github.com/rakutentech/cf-tools/blob/a8e2e93/cf-bg-restart.sh#L72. You may want to update also.

**Comment by ArthurHlt** [comment]
Hey @giner thanks for the PR and sorry for the delay

You have made some changes likes for venerable suffix and did some refactoring, this was actually hard to review everything (the part about droplet wasn't clear but i've found that it was affecting your modification)

Anyway thanks for the work, just note those few advices to help next time for understanding and reading

**Pull request merged**